traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/ds/LetsEncryptDnsChallengeWatcher.java

```diff
@@ -0,0 +1,173 @@
+                    } else {
+                        nameSb.append('.');
+                        nameSb.append(label);
+                    }
+                }
+
+                ArrayNode staticDnsEntriesNode;
+
+                if (deliveryServiceConfig.findValue("staticDnsEntries") != null) {
+                    staticDnsEntriesNode = (ArrayNode) deliveryServiceConfig.findValue("staticDnsEntries");
+                } else {
+                    staticDnsEntriesNode = mapper.createArrayNode();
+                }
+
+                if (challenge.getRecord().isEmpty()) {
+                    for (int i = 0; i < staticDnsEntriesNode.size(); i++) {
+                        if (staticDnsEntriesNode.get(i).get("name").equals("_acme-challenge")) {
+                            staticDnsEntriesNode.remove(i);
+                        }
+                    }
+                } else {
+
+                    final ObjectNode newChildNode = mapper.createObjectNode();
+                    newChildNode.put("type", "TXT");
+                    newChildNode.put("name", nameSb.toString());
+                    newChildNode.put("value", challenge.getRecord());
+                    newChildNode.put("ttl", 10);
+
+                    staticDnsEntriesNode.add(newChildNode);
+                }
+
+                deliveryServiceConfig.set("staticDnsEntries", staticDnsEntriesNode);
+                deliveryServicesNode.set(dsLabel, deliveryServiceConfig);
+
+            });
+
+            final ObjectNode statsNode = (ObjectNode) mostRecentConfig.get("stats");
+            statsNode.put("date", Instant.now().toEpochMilli() / 1000L);
+
+            final ObjectNode fullConfig = (ObjectNode) mostRecentConfig;
+            fullConfig.set(ConfigHandler.deliveryServicesKey, deliveryServicesNode);
+            fullConfig.set("stats", statsNode);
+
+            try {
+                configHandler.processConfig(fullConfig.toString());
+            } catch (JsonParseException | JsonUtilsException jsonError) {
+                LOGGER.error("error processing config: " + jsonError.getMessage());
+            }
+
+            return true;
+        } catch (Exception e) {
+            LOGGER.warn("Failed updating dns challenge txt record with data from " + dataBaseURL + ": " + e.getMessage());
+        }
+
+        return false;
+    }
+
+    @Override
+    protected boolean verifyData(final String data) {
+        try {
+            final ObjectMapper mapper = new ObjectMapper(new JsonFactory());
+            mapper.readValue(data, new TypeReference<HashMap<String, List<LetsEncryptDnsChallenge>>>() { });
+            return true;
+        } catch (Exception e) {
+            LOGGER.warn("Failed to build dns challenge data while verifying");
+        }
+
+        return false;
+    }
+
+    @Override
+    public String getWatcherConfigPrefix() {
+        return "dnschallengemapping";
+    }
+
+    @SuppressWarnings("PMD.AppendCharacterWithChar")
```

Comment:
This is a simple enough fix, just use single-quotes. No need to suppress it.

Reply:
done